src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -226,8 +227,10 @@ void uct_tcp_sockcm_ep_handle_event_status(uct_tcp_sockcm_ep_t *ep,
 
     /* if the ep is on the server side but uct_ep_create wasn't called yet,
      * destroy the ep here since uct_ep_destroy won't be called either */
-    if ((ep->state & UCT_TCP_SOCKCM_EP_ON_SERVER) &&
-        !(ep->state & UCT_TCP_SOCKCM_EP_SERVER_CREATED)) {
+    if ((ep->state & (UCT_TCP_SOCKCM_EP_ON_SERVER |
+                      UCT_TCP_SOCKCM_EP_SERVER_CREATED |
+                      UCT_TCP_SOCKCM_EP_SERVER_CONN_REQ_CB_INVOKED)) ==
```

Comment:
@dmitrygx isn't it enough to check only that UCT_TCP_SOCKCM_EP_SERVER_CREATED isn't set? if we get here and only UCT_TCP_SOCKCM_EP_SERVER_CONN_REQ_CB_INVOKED is set, it means that UCT_TCP_SOCKCM_EP_SERVER_CREATED isn't set and so we need to close the internal ep.

Reply:
TCP_SOCKCM received a connection request from a client, TCP_SOCKCM allocates internal UCT EP that is returned to a user as a connextion_request. Then, a user on a client side forcibly closes it's UCT EP, so a server side detects connection reset and invokes error handling scenario which destroyed created UCT EP. Then, a user on a server side creates UCT EP by calling `uct_ep_create()` and specifying connection_request got from listener. So, `uct_ep_create()` dereferenced connection_request (i.em UCT EP), but this is invalid anymore.
The idea of the PR is to not remove UCT EP when connection_requset was provided to a user, just mark UCT EP as a FAILED one. And then, check FAILED flag when creating UCT EP using connection_request from a user.

Reply:
@dmitrygx thanks for the explanation!
If we are now not closing the tcp_sockcm internal ep here, i.e. before uct_ep_create was called, then the doc comment isn't correct anymore. Can you please update it?

Reply:
done